pkg/bff/collaborators.go

```diff
@@ -34,13 +39,33 @@ func (s *Server) AddCollaborator(c *gin.Context) {
 		return
 	}
 
-	// Add the collaborator to the organization
-	if err = org.AddCollaborator(collaborator); err != nil {
-		log.Error().Err(err).Msg("could not add new collaborator to organization")
+	// Make sure the collaborator is valid for storage
+	if err = collaborator.Validate(); err != nil {
+		log.Warn().Err(err).Msg("invalid collaborator in request")
 		c.JSON(http.StatusBadRequest, api.ErrorResponse(err))
 		return
 	}
 
+	// Don't overwrite an existing collaborator
+	id := collaborator.Key()
+	if _, ok := org.Collaborators[id]; ok {
+		log.Warn().Str("collabID", id).Msg("collaborator already exists")
+		c.JSON(http.StatusConflict, api.ErrorResponse("collaborator already exists"))
+		return
+	}
+
+	// Make sure the record has a created timestamp
+	if collaborator.CreatedAt == "" {
+		collaborator.CreatedAt = time.Now().UTC().Format(time.RFC3339Nano)
+	}
+
+	// Add the collaborator to the organization
+	if org.Collaborators == nil {
+		org.Collaborators = make(map[string]*models.Collaborator)
+	}
+	org.Collaborators[id] = collaborator
+
+	// TODO: We can search the email address in Auth0 to see if the user already exists
 	// TODO: Send invite/verification email to the collaborator
```

Comment:
Do we have stories for these TODOs?

Reply:
I'll create them before closing this story.